argoverse/evaluation/detection_utils.py

```diff
@@ -0,0 +1,175 @@
+increased interpretability of the error modes in a set of detections.
+
+"""
+
+from enum import Enum, auto
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+from scipy.spatial.distance import cdist
+from scipy.spatial.transform import Rotation as R
+
+from argoverse.data_loading.object_label_record import ObjectLabelRecord
+from argoverse.evaluation.eval_tracking import get_orientation_error_deg
+from argoverse.utils.transform import quat_argo2scipy_vectorized
+
+
+class SimFnType(Enum):
+    CENTER = auto()
+
+
+class DistFnType(Enum):
+    TRANSLATION = auto()
+    SCALE = auto()
+    ORIENTATION = auto()
+
+
+class InterpType(Enum):
+    ALL = auto()
+
+
+class FilterMetric(Enum):
+    EUCLIDEAN = auto()
+
+
+def filter_instances(
+    instances: List[ObjectLabelRecord],
+    target_class: str,
+    filter_metric: FilterMetric = FilterMetric.EUCLIDEAN,
+    max_dist: float = 50.0,
+) -> np.ndarray:
+    """Filter the GT annotations based on a set of conditions (e.g., distance from egovehicle).
+
+    Args:
+        instances: The instances to be filtered (N,).
+        target_class: The name of the class of interest.
+        filter_metric: The range metric used for filtering.
+        max_dist: The maximum distance for range filtering.
+
+    Returns:
+        The filtered annotations.
+    """
+    instances = np.array([instance for instance in instances if instance.label_class == target_class])
+
+    if filter_metric == FilterMetric.EUCLIDEAN:
+        centers = np.array([dt.translation for dt in instances])
+        filtered_annos = np.array([])
+
+        if centers.shape[0] > 0:
+            dt_dists = np.linalg.norm(centers, axis=1)
+            filtered_annos = instances[dt_dists < max_dist]
+    else:
+        raise NotImplementedError("This filter metric is not implemented!")
+    return filtered_annos
+
+
+def get_ranks(dts: List[ObjectLabelRecord]) -> Tuple[np.ndarray, np.ndarray]:
+    """Get the rankings for the detections.
+
+    Args:
+        dts: Detections (N,).
+
+    Returns:
+        scores: The detection scores (N,).
+        ranks: The ranking for the detections (N,).
+
+    """
+    scores = np.array([dt.score for dt in dts])
+    ranks = scores.argsort()[::-1]
+    return np.expand_dims(scores, 1)[ranks], ranks
+
+
+def interp(prec: np.ndarray, method: InterpType = InterpType.ALL) -> np.ndarray:
+    """Interpolate the precision over all recall levels.
+
+    Args:
+        prec: Precision at all recall levels (N,).
+        method: Accumulation method.
+
+    Returns:
+        prec_interp: Interpolated precision at all recall levels.
+    """
+    if method == InterpType.ALL:
+        prec_interp = np.maximum.accumulate(prec[::-1])[::-1]
+    else:
+        raise NotImplemented("This interpolation method is not implemented!")
+    return prec_interp
+
+
+def compute_match_matrix(dts: List[ObjectLabelRecord], gts: List[ObjectLabelRecord], metric: SimFnType) -> np.ndarray:
```

Comment:
might prefer the term affinity matrix, rather than match matrix, to indicate that no match has been selected yet